### ISSUE TITLE
Amber build fails to build ligand correctly

### ISSUE THREAD

**Issue by GerardBCN** [Acellera/htmd#179]
When building a protein-ligand system with AMBER, the intermediate input.pdb generated contains something like this:

```
TER
ATOM  10119  O1  MOL A   1      -9.968   2.144   5.023  1.002383.0      L    O  
TER
ATOM  10120  O   WAT W1871     -32.392 -27.731 -37.007  1.002384.0      WT0     
TER
ATOM  10121  C1  MOL A   1      -8.797   1.504   4.783  1.002385.0      L    C  
TER
ATOM  10122  H1  WAT W1871     -32.785 -28.464 -37.553  1.002386.0      WT0     
TER
ATOM  10123  C2  MOL A   1      -7.612   2.189   4.553  1.002387.0      L    C  
TER
ATOM  10124  H2  WAT W1871     -32.277 -27.070 -37.770  1.002388.0      WT0     
TER
ATOM  10125  C3  MOL A   1      -8.811   0.105   4.732  1.002389.0      L    C  
TER
ATOM  10126  O   WAT W1872      -8.563 -24.042 -32.577  1.002390.0      WT0     
TER
ATOM  10127  O2  MOL A   1      -9.982  -0.541   4.986  1.002391.0      L    O  
TER
ATOM  10128  H1  WAT W1872      -8.204 -23.312 -32.121  1.002392.0      WT0     
TER
ATOM  10129  C4  MOL A   1      -7.642  -0.574   4.439  1.002393.0      L    C  
ATOM  10130  C5  MOL A   1      -6.468   0.114   4.206  1.002393.0      L    C  
TER
ATOM  10131  H2  WAT W1872      -9.226 -23.713 -33.150  1.002394.0      WT0     
ATOM  10132  O   WAT W1873     -25.827 -28.810 -32.200  1.002395.0      WT0     
TER
ATOM  10133  C6  MOL A   1      -6.439   1.507   4.260  1.002396.0      L    C  
ATOM  10134  C7  MOL A   1      -5.155   2.259   3.977  1.002396.0      L    C  
TER
ATOM  10135  H1  WAT W1873     -26.752 -29.051 -31.870  1.002397.0      WT0     
ATOM  10136  H2  WAT W1873     -25.352 -28.420 -31.435  1.002397.0      WT0     
TER
ATOM  10137  O3  MOL A   1      -5.308   3.211   2.927  1.002398.0      L    O  
ATOM  10138  C8  MOL A   1      -4.683   3.029   5.197  1.002398.0      L    C  
TER
ATOM  10139  O   WAT W1874     -26.317 -20.529 -16.427  1.002399.0      WT0     
ATOM  10140  H1  WAT W1874     -25.921 -21.424 -16.374  1.002399.0      WT0     
TER
ATOM  10141  N1  MOL A   1      -3.460   3.810   4.935  1.002400.0      L    N  
ATOM  10142  H1  MOL A   1     -10.104   2.218   5.970  1.002400.0      L    H  
TER
ATOM  10143  H2  WAT W1874     -27.221 -20.596 -16.215  1.002401.0      WT0     
TER
ATOM  10144  H2  MOL A   1      -7.602   3.278   4.603  1.002402.0      L    H  
TER
ATOM  10145  O   WAT W1875     -26.492 -14.481 -35.034  1.002403.0      WT0     
TER
ATOM  10146  H3  MOL A   1     -10.109  -0.611   5.935  1.002404.0      L    H  
TER
ATOM  10147  H1  WAT W1875     -26.182 -13.970 -34.243  1.002405.0      WT0     
ATOM  10148  H2  WAT W1875     -26.171 -15.351 -34.857  1.002405.0      WT0     
TER
ATOM  10149  H4  MOL A   1      -7.646  -1.663   4.391  1.002406.0      L    H  
ATOM  10150  H5  MOL A   1      -5.560  -0.445   3.979  1.002406.0      L    H  
TER
ATOM  10151  O   WAT W1876     -29.229 -10.507 -17.416  1.002407.0      WT0     
TER
ATOM  10152  H6  MOL A   1      -4.425   1.525   3.634  1.002408.0      L    H  
TER
ATOM  10153  H1  WAT W1876     -29.927  -9.824 -17.209  1.002409.0      WT0     
TER
ATOM  10154  H7  MOL A   1      -5.430   2.751   2.093  1.002410.0      L    H  
TER
ATOM  10155  H2  WAT W1876     -29.010 -10.878 -16.559  1.002411.0      WT0     
ATOM  10156  O   WAT W1877     -16.580  -6.760 -22.793  1.002412.0      WT0     
TER
ATOM  10157  H8  MOL A   1      -5.457   3.763   5.421  1.002413.0      L    H  
TER
ATOM  10158  H1  WAT W1877     -17.209  -6.843 -22.063  1.002414.0      WT0     
TER
ATOM  10159  H9  MOL A   1      -4.402   2.288   5.945  1.002415.0      L    H  
ATOM  10160  H10 MOL A   1      -2.570   3.340   4.845  1.002415.0      L    H  
TER
ATOM  10161  H2  WAT W1877     -16.328  -7.703 -22.993  1.002416.0      WT0     
ATOM  10162  O   WAT W1878     -16.054 -31.549 -17.284  1.002417.0      WT0     
TER
ATOM  10163  H11 MOL A   1      -3.512   4.815   4.845  1.002418.0      L    H  
ATOM  10164  H12 MOL A   1      -2.585   4.315   4.935  1.002418.0      L    H  
TER
```
which later makes tleap to fail and try to regenerate one ligand from each of the TER-separated atoms from the same ligand. Am I doing something wrong, perhaps? I defined the ligand as segid L.

**Comment by stefdoerr** [comment]
Why are there waters in your ligand?

**Comment by GerardBCN** [comment]
There are none, actually. Notice how the segids are well set.

**Comment by GerardBCN** [comment]
Stefan suggests it may be related to the capping, @j3mdamas

**Comment by j3mdamas** [comment]
I'll need to see this when I'm back. I haven't push my last changes to the
amber capping. But I recall Stefan changing ligand treatment in amber,
didn't you Stefan?

**Comment by stefdoerr** [comment]
Yeah but the capping happens before that

**Comment by j3mdamas** [comment]
Ok I'll have to check it then

**Comment by stefdoerr** [comment]
Gerard can you provide a zip with the files required to reproduce it?

**Comment by GerardBCN** [comment]
http://pub.htmd.org/amber_failing.zip.zip

**Comment by tonigi** [comment]
It's tricky. It does not occur in the debugger, for example.  No clue for now.

**Comment by tonigi** [comment]
Should be fixed in d908c5e1f0418c8b9d52063b9b7b9078b48cefb0

**Comment by j3mdamas** [comment]
Cool Toni, thanks!
